### PR TITLE
internal/web3ext: make whisper v5 methods work

### PR DESCRIPTION
**Pull request by gballet**
Changes happening in the last few months have introduced discrepancies between the Go functions and their JS counterpart. This PR makes sure that everything is up to date. This PR:

- Updates the whisper-related part of `/internal/jsre/deps` to the state it currently is in the web3.js repository. No other part has been imported to ensure that there are no conflicts.
- Changes the name of two functions in `web3ext.go`, that were not updated when the names got changed.

### CODE REVIEW

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: a824ef756e0ac025227e057657037a852ce9f1a0
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

File: internal/web3ext/web3ext.go [hunk 1]
```diff
@@ -532,13 +532,13 @@ web3._extend({
...
+			name: 'setMaxMessageSize',
+			call: 'shh_setMaxMessageSize',
 			params: 1
 		}),
 		new web3._extend.Method({
-			name: 'setMinimumPoW',
-			call: 'shh_setMinimumPoW',
+			name: 'setMinPoW',
+			call: 'shh_setMinPoW',
 			params: 1
```
Comment (bas-vk) [inline internal/web3ext/web3ext.go]: web3etx is reserved for functionality not offered by web3. Now that web3 has whisper support most or maybe all ext shh methods/properties can be removed.

Comment (gballet) [reply internal/web3ext/web3ext.go]: That's good to know, thanks. I have checked which functions are indeed present, and it seems that only `getFilterMessages`, `deleteMessageFilter` and `publicKey` are missing. I'm waiting for Fabian to discuss the web3.js update, but if I can add those functions then indeed they could be removed from this file.

Comment (gballet) [reply internal/web3ext/web3ext.go]: @bas-vk so actually, in this repo the web3.js doesn't work (it uses the http rpc interface that needs to be started explicitly), and it's version 0.20.1 which doesn't work anyway. I wanted to upgrade to 1.0.0 but Fabian says that this won't work. So we still need the web3ext functions for now.

Comment (gballet) [reply internal/web3ext/web3ext.go]: Alright, I managed to get 0.20.1 to work, so I updated it and now the api does work, I am able to send and receive messages. So as you suggested, @bas-vk , I removed the web3ext functions.

File: internal/jsre/deps/web3.js [hunk 2]
```diff
@@ -5902,35 +6065,28 @@ var eth = function () {
...
+            call: 'shh_newMessageFilter',
+            params: 1
+        }),
+        new Method({
+            name: 'uninstallFilter',
+            call: 'shh_deleteMessageFilter',
+            params: 1
+        }),
+        new Method({
+            name: 'getLogs',
```
Comment (bas-vk) [inline internal/jsre/deps/web3.js]: Not sure about `getLogs`, its name is not something that I would expect to retrieve messages.

For `eth` we don't offer these raw filter methods. Web3 has its own [filter](https://github.com/ethereum/wiki/wiki/JavaScript-API#web3ethfilter) mechanism that calls these methods under the hood. I tried your branch with the updated web3 and I looks like this is not implemented for whisper, @frozeman?

Comment (gballet) [reply internal/jsre/deps/web3.js]: According to @frozeman on gitter:
> Yes it’s only used internal and the name is to have one in both namespaces. But as this is the legacy web3.js I don’t want to change it

These are actually not the functions that are available to the user, they are defined in `watches.js` (not `shh.js`), they are used by the filter code and get overwritten when `Shh` gets instantiated. So it's perfectly safe to leave them here, and actually important according to Fabian.

**Comment by frozeman** [comment]
Though im not so sure as its mainly this commit on the patch version from 0.20.1 to 0.20.2:
https://github.com/ethereum/web3.js/commit/f74d60c1711d62520783311db55ef2ae80a71351

**Pull request merged**